Increase visibility for ADAMContext.sc and/or getFs... methods

The Hadoop configuration from the SparkContext associated with an ADAMContext needs to be accessible for subclasses to access the FileSystem
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] .../org/dishevelled/bio/adam/BiojavaAdamContext.java:[136,52]
  sc has private access in org.bdgenomics.adam.rdd.ADAMContext
```

Alternately, we could increase the visibility of the `getFs...` methods from `private[rdd]` to `protected`.

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala
@@ -406,7 +406,7 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
    *
    * @throws FileNotFoundException if the path does not match any files.
    */
-  private def getFiles(path: Path, fs: FileSystem): Array[Path] = {
+  protected def getFiles(path: Path, fs: FileSystem): Array[Path] = {
 
     // elaborate out the path; this returns FileStatuses
     val paths = if (fs.isDirectory(path)) fs.listStatus(path) else fs.globStatus(path)
@@ -435,7 +435,7 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
    *
    * @throws FileNotFoundException if the path does not match any files.
    */
-  private[rdd] def getFsAndFiles(path: Path): Array[Path] = {
+  protected def getFsAndFiles(path: Path): Array[Path] = {
 
     // get the underlying fs for the file
     val fs = Option(path.getFileSystem(sc.hadoopConfiguration)).getOrElse(
@@ -457,7 +457,7 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
    *
    * @throws FileNotFoundException if the path does not match any files.
    */
-  private def getFsAndFilesWithFilter(filename: String, filter: PathFilter): Array[Path] = {
+  protected def getFsAndFilesWithFilter(filename: String, filter: PathFilter): Array[Path] = {
 
     val path = new Path(filename)
 
@@ -475,7 +475,7 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     }
 
     // the path must match at least one file
-    if (paths.isEmpty) {
+    if (paths == null || paths.isEmpty) {
       throw new FileNotFoundException(
         s"Couldn't find any files matching ${path.toUri}"
       )

--- a/adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala
+++ b/adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala
@@ -427,23 +427,22 @@ class ADAMContextSuite extends ADAMFunSuite {
     reads.saveAsParquet(outputPath.replace(".adam", ".2.adam"))
 
     val paths = new Path(outputPath.replace(".adam", "*.adam") + "/*")
-    assert(sc.getFsAndFiles(paths).size > 2)
 
     val reloadedReads = sc.loadParquetAlignments(outputPath.replace(".adam", "*.adam") + "/*")
     assert((2 * reads.rdd.count) === reloadedReads.rdd.count)
   }
 
   sparkTest("bad glob should fail") {
-    val inputPath = testFile("small.sam")
+    val inputPath = testFile("small.sam").replace(".sam", "*.sad")
     intercept[FileNotFoundException] {
-      sc.getFsAndFiles(new Path(inputPath.replace(".sam", "*.sad")))
+      sc.loadAlignments(inputPath)
     }
   }
 
   sparkTest("empty directory should fail") {
-    val outputPath = tmpLocation()
+    val inputPath = tmpLocation()
     intercept[FileNotFoundException] {
-      sc.getFsAndFiles(new Path(outputPath))
+      sc.loadAlignments(inputPath)
     }
   }
 
@@ -491,4 +490,3 @@ class ADAMContextSuite extends ADAMFunSuite {
     })
   }
 }
-

